addons/survey/static/src/js/survey_form.js

```diff
@@ -55,6 +55,10 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
                 self.preventEnterSubmit = true;
             }
             self._initSessionManagement();
+
+            // Needs global selector as the progress is not within the survey form, but needs to be
+            // updated at the same time
+            self.$surveyProgress = $('.o_survey_progress_wrapper');
```

Comment:
[C5] A bit strange to update html outside of the widget's scope. I suppose having another level (survey_page with a form and a progress) would require too much refactoring ?

Reply:
The problem is that the progress / navigation are now included in the generic "survey.layout" template.

So the widget would have to be plugged really high to include all that (probably something like "#wrap") which is most likely not a good idea.

I agree that this needs to be refactored.
We need to sit down and think about how we could organize those templates better in terms of structure.